src/commands/createStaticWebApp/createStaticWebApp.ts

```diff
@@ -22,6 +22,16 @@ export async function createStaticWebApp(context: IActionContext & Partial<ICrea
         node = await ext.tree.showTreeItemPicker<SubscriptionTreeItem>(SubscriptionTreeItem.contextValue, context);
     }
 
+    const folder: WorkspaceFolder = await getWorkspaceFolder(context);
+    const gitWorkspaceState: GitWorkspaceState = await getGitWorkspaceState(context, folder.uri);
+    // verifyRepoForCreation will set gitWorkspaceState.repo, but typing will yell if we don't set it here
```

Comment:
What about changing `verifyRepoForCreation`'s return type to something like this? 
```ts 
type VerifiedGitWorkspaceState = GitWorkspaceState & { repo: Repository };
```
Then you could just set `gitWorkspaceState = await verifyRepoForCreation(...);`